strax/mailbox.py

```diff
@@ -467,13 +493,14 @@ def divide_outputs(source,
     # TODO: this code duplicates exception handling and cleanup
     # from Mailbox.send_from! Can we avoid that somehow?
     i = 0
+    counter = 0
```

Comment:
Perhaps make it a bit more explicit (I would have to figure out the difference between `counter` and `i` whereas ` counter` is mostly a flag right)?
```suggestion
    at_least_one_flowing = False
```
If I understand, this is the key fix to the deadlock? If so perhaps add a few comments / ref to this PR